### PR TITLE
Tweak translation scenarios to allow Same labels per context

### PR DESCRIPTION
**Pull request by daneryl**
- Fixed Bug 1: Old translation keys not deleted when rename skipped due to collision
- Fixed Bug 2: Shared keys (template name + property label) incorrectly deleted during property renames
- Fixed Bug 3: Context protection now prevents all deletions (automatic + explicit)

Changes to UpsertTranslationsService:
- Modified updateContext() to automatically delete old keys from all renames
- Added deduplicateKeyChanges() to skip renames when old key still needed in context
- Added calculateKeysToRemove() to collect all keys to delete while protecting keys in valueChanges
- Context (valueChanges) is now source of truth - keys in final context are always protected

Changes to LegacyTemplatesTranslationService:
- Simplified caller code by removing redundant deletedLabels tracking
- Now only passes explicitly deleted properties in keysToDelete
- Relies on updateContext() to handle rename deletions automatically

Added comprehensive test coverage for edge cases including collision handling, shared keys, and context protection.

### CODE REVIEW

**Review by daneryl — COMMENTED** [review]
**Re: Chained rename concern (Comment 1)**

This behavior is intentional. In the scenario A→B, B→C where B exists:
- A→B is blocked (collision)
- B→C executes  
- A is deleted because it's not in valueChanges (final context)
- B remains as a shared key (e.g., template name + property label)

valueChanges is the source of truth. If A should become B, then B would be in valueChanges and A would not be deleted. The deletion means A is genuinely not needed in the final context.

Since this review, we've refactored into TranslationContextModel with 27 tests covering shared keys and collision handling.

**Review by daneryl — COMMENTED** [review]
**Re: Performance concern - Loading all translations (Comment 2)**

✅ Fixed in refactoring (commits 2da67c00c0 and 579e038000).

- Extracted logic into TranslationContextModel domain model
- Model receives translations via translationsDS.getContext() once and reuses them
- Eliminated separate getExistingKeysInContext() query
- Collision detection now happens in-memory

Result: Reduced from 5-7 DB operations to 2 (1 fetch + 1 bulk write).

**Review by daneryl — COMMENTED** [review]
**Re: Unused parameter (Comment 3)**

✅ Fixed in refactoring.

Moved calculateKeysToRemove() into TranslationContextModel where it only takes parameters it actually uses. See TranslationContextModel.ts lines 155-171.

File: app/api/i18n.v2/services/UpsertTranslationsService.ts [hunk 1]
```diff
@@ -130,4 +173,53 @@ export class UpsertTranslationsService {
...
+    const deduplicated: { [oldKey: string]: string } = {};
+
+    Object.entries(keyChanges).forEach(([oldKey, newKey]) => {
+      const targetExists = existingKeys.has(newKey);
+      const isNoOp = oldKey === newKey;
+      const oldKeyStillNeeded = !!valueChanges[oldKey];
+
+      if ((!targetExists || isNoOp) && !oldKeyStillNeeded) {
+        deduplicated[oldKey] = newKey;
+      }
```
[the remark this answers is not in the document]
Comment (daneryl) [reply app/api/i18n.v2/services/UpsertTranslationsService.ts]: This behavior is intentional. In the scenario A→B, B→C where B exists:
- A→B is blocked (collision)
- B→C executes  
- A is deleted because it's not in valueChanges (final context)
- B remains as a shared key (e.g., template name + property label)

valueChanges is the source of truth. If A should become B, then B would be in valueChanges and A would not be deleted.

File: app/api/i18n.v2/services/UpsertTranslationsService.ts [hunk 2]
```diff
@@ -130,4 +173,53 @@ export class UpsertTranslationsService {
...
+
+  /**
+   * Gets all existing translation keys within a specific context.
+   * Deduplication only affects keys within the same context.
+   * Different contexts can have the same keys without conflict.
+   */
+  private async getExistingKeysInContext(contextId: string): Promise<Set<string>> {
+    const translations = await this.translationsDS.getByContext(contextId).all();
+    return new Set(translations.map(t => t.key));
+  }
```
[the remark this answers is not in the document]
Comment (daneryl) [reply app/api/i18n.v2/services/UpsertTranslationsService.ts]: ✅ Fixed in refactoring (commits 2da67c00c0 and 579e038000).

- Extracted logic into TranslationContextModel domain model
- Model receives translations via translationsDS.getContext() once and reuses them
- Eliminated separate getExistingKeysInContext() query
- Collision detection now happens in-memory

Result: Reduced from 5-7 DB operations to 2 (1 fetch + 1 bulk write).

File: app/api/i18n.v2/services/UpsertTranslationsService.ts [hunk 3]
```diff
@@ -130,4 +173,53 @@ export class UpsertTranslationsService {
...
+
+    return deduplicated;
+  }
+
+  private calculateKeysToRemove(
+    originalKeyChanges: { [oldKey: string]: string },
+    deduplicatedKeyChanges: { [oldKey: string]: string },
+    keysToDelete: string[],
+    valueChanges: { [key: string]: string }
+  ): string[] {
```
[the remark this answers is not in the document]
Comment (daneryl) [reply app/api/i18n.v2/services/UpsertTranslationsService.ts]: ✅ Fixed in refactoring.

Moved calculateKeysToRemove() into TranslationContextModel where it only takes parameters it actually uses. See TranslationContextModel.ts lines 155-171.
